### PR TITLE
Testing to add spring boot 2.3.x version support

### PR DESCRIPTION
**Pull request by moarychan**
This PR is for testing the azure spring modules based on the spring boot 2.3.x version in a new service directory.

#### Changelog:
- Add the new service directory `spring-2-3`.
- Check out from tag `azure-spring-boot_3.2.0` which is based on spring boot version `2.3.5.RELEASE` version.
- Upgrade spring boot version to `2.3.10.RELEASE`.
- Increase azure spring modules version.
- Another PR https://github.com/Azure/azure-sdk-for-java/pull/21385 to add live test for service directory `spring-2-3`.

### Points to consider when adapting `Spring boot 2.3.x`
- Need new tags hosted, more and more version tags will be added to the dependency management file.
- [TBC] Can only include one coordinate dependency in 'jacoco-test-coverage' module.
- [TBC, solved] Upgrade pom_file_version_scanner.ps1 script to support non-grouopid-artfactId version tag.
- [TBC, solved] Upgrade maven '-pl' parameter to support different service directory with the same group id and artfactid.
  `mvn -f pom.xml xxx -Drevapi.skip=true -pl ,com.azure.spring:azure-identity-spring`
   --> 
  `mvn -f pom.xml xxx -Drevapi.skip=true -pl ,./sdk/spring/azure-identity-spring`
  `mvn -f pom.xml xxx -Drevapi.skip=true -pl ,./sdk/spring-2-3/azure-identity-spring`
- [TBC] how to ship a new version for azure-spring-boot-bom based on spring boot 2.3.x?
- Add a new version number for all spring samples, IT test modules, and the root pom of the service directory.
- [TBC, solved] Upgrade script `generate_from_source_pom.py` to support project list based on relative path.
- Projects with multi POM files need to be split into a single, such as `azure-spring-boot-test-keyvault`.
- The current version of Azure SDK that Spring Boot low version depends on should not be the Beta version(`azure-spring-data-cosmos`).
- [TBC] How to fix the third-party dependency conflicts? e.g, Spring Boot and Azure SDK depend on different versions of the same third-party module at the same time.
- [TBC] Version number planning for other non-major versions.

### CODE REVIEW

**Comment by saragluna** [comment]
Hi @mitchdenny, this is a try to support Spring versions side by side https://github.com/Azure/azure-sdk-for-java/issues/15512. Could you please take a look at this.

**Comment by saragluna** [comment]
@mitchdenny to change the artifact names having a spring version identifier will not work for us, since we've released our libs for Spring Boot 2.3.x and 2.4.x. So we're giving multiple directories a try now, to see whether it is feasible this way.

**Comment by mitchdenny** [comment]
I can't review this PR anymore because the number of files is too large (causes the browser to hang). Perhaps that is a sign that we should be looking at making these changes in more digestible chunks?

**Review by mitchdenny — CHANGES_REQUESTED** [review]
This PR can't be merged yet since we haven't resolved the "should we" question. But there are probably approaches that we'd want to take to make a more general solution to side-by-side versioning IF we want to take this approach.

It certainly needs to be broken down into smaller charges. A PR with 1296 changes is probably not viable to review sensibly.

File: sdk/spring-2-3/AzureSpringCodeStyle.xml [hunk 1]
```diff
@@ -0,0 +1,376 @@
+<code_scheme name="AzureSpringCodeStyle20200910" version="173">
```
[the remark this answers is not in the document]
Comment (JonathanGiles) [reply sdk/spring-2-3/AzureSpringCodeStyle.xml:1]: I suspect this file should just be removed before the PR is merged.

**Comment by moarychan** [comment]
@mitchdenny , sorry for confusing you, I have put this PR as a draft and it should not be merged to master. Thanks for your suggestion, I will continue to test other questions.

**Pull request closed without merging**